exporters/otlp/otlpmetric/otlpmetrichttp/client.go

```diff
@@ -166,8 +166,7 @@ func (c *client) UploadMetrics(ctx context.Context, protoMetrics *metricpb.Resou
 			if _, err := io.Copy(&respData, resp.Body); err != nil {
 				return err
 			}
-
-			if respData.Len() != 0 {
```

Comment:
Should this not continue to also check the response length is greater than zero?

Reply:
If the server returns `Content-Type: application/x-protobuf` it to have the content.

Reply:
I don't think that is a guarantee. An empty response if still going to be valid with those header values.

I would continue to include this guard.

Reply:
I added https://github.com/open-telemetry/opentelemetry-go/pull/4719/commits/46d0335ea63438d29d107d401c2a8edf773d5027

It occurs that this condition is even covered by existing test suite 🎉